substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -1283,10 +1283,19 @@ pub mod pallet {
 			payee: RewardDestination<T::AccountId>,
 		) -> DispatchResult {
 			let controller = ensure_signed(origin)?;
-			let ledger = Self::ledger(Controller(controller))?;
+			let ledger = Self::ledger(Controller(controller.clone()))?;
+
+			// Ensure that the `Controller` variant is not assigned.
+			#[allow(deprecated)]
+			let payee_final = if payee == RewardDestination::Controller {
```

Comment:
Should we just error out if someone passes RewardDestination::Controller? I think it will be more cleaner.

Reply:
Yes, either way would work. I have added the error in the latest commit